### ISSUE TITLE
Shellexec not recognising any variable (title formatting) I've tried

### ISSUE THREAD

**Issue by fabertawe** [DeaDBeeF-Player/deadbeef#2191]
### Steps to reproduce the problem
Configure a custom shell command

### What's going on? Describe the problem in as much detail as possible.
I'm coming back to DeaDBeeF after some time (since 0.7?) and a custom shell command I used to use is no longer working. Instead of appending track details to a playlist on disk it now saves the literal variable name -
`if [ ! -f /dev/shm/playlist.m3u ]; then echo '#EXTM3U' > /dev/shm/playlist.m3u; fi; echo "#EXTINF:%@:DURATION@,$meta(%title%),%title,%title%,$meta(%title),$meta(title),%path% - %filename% - %D" >> /dev/shm/playlist.m3u`
results in this being saved to the file as
`#EXTINF:%@:DURATION@,(%title%),%title,%title%,(%title),(title),%path% - %filename% - %D`

Any variable (I tried) is now literally interpreted as plain text, e.g. xdg-open %D
`xdg-open: file '%D' does not exist`

### Information about the software:
This is on a fully up to date Arch Linux with versions 1.8.2 and devel.

**Comment by Alexey-Yakovenko** [comment]
Shellexec doesn't support "new" title formatting. It's still using 1.0 title formatting, which is documented in wiki.

**Comment by Alexey-Yakovenko** [comment]
duplicate of #1556

**Comment by fabertawe** [comment]
But it's also not recognising 1.0 title formatting, as indicated above by the xdg-open example.

**Comment by Alexey-Yakovenko** [comment]
Sorry, I didn't notice the actual problem behind the irrelevant info.

**Comment by fabertawe** [comment]
I've just discovered after testing again, xdg-open works for me if I untick "Generic (Main Menu)" in the custom shell command "Edit Command" and use it from the right click context menu. It doesn't work if the command is run from the main menu.

Should this one be closed and I open a new report for this?

**Comment by Alexey-Yakovenko** [comment]
I think that everything works the way it's supposed to work, and opening a bug for each thing you didn't quite figure out yet would not get us far. Shellexec implementation is far from perfect and very old. But I'm pretty sure that shellexec main menu items weren't supposed to handle title formatting at all.

**Comment by fabertawe** [comment]
> But I'm pretty sure that shellexec main menu items weren't supposed to handle title formatting at all.

In that case there is no bug and it was all a lack of understanding on my part. Thanks.

**Comment by Alexey-Yakovenko** [comment]
The only thing that is suspicious though is that you claimed the command worked in the past.
If this is the case -- it's supposed to still work, although the command you provided in the summary is not supposed to work in the previous or current versions.

**Comment by fabertawe** [comment]
I'll be honest - I can't remember exactly what I had working previously but obviously spent some time implementing the custom commands when I was using DeaDBeeF originally. That one I posted is a bit of a mess actually and doesn't do what I intended. I've corrected it my end and it now works very nicely from the right-click context menu.

Maybe just a mention in the FAQ/Wiki will be sufficient regarding no TF from the main menu with custom commands.

Apologies if I've wasted your time with this. I'm now using DeaDBeeF extensively and really appreciate your work.
